q2_fondue/sequences.py

```diff
@@ -238,6 +237,20 @@ def _write_empty_casava(read_type, casava_out_path):
             pass
 
 
+def _rewrite_fastq(file_in: str, file_out: str):
+    with open(file_in, 'r') as fin, open(file_out, 'w') as fout:
+        for line in fin.readlines():
+            fout.write(line.strip() + '\n')
```

Comment:
`fh.readlines()` will load an entire file into a list, usually you would just iterate over the file.
Also, I'm not sure if you need to strip the newline and add it back. Are you running into extra whitespace characters at the start or end of a line that you are specifically avoiding?

If not, then the fast way would be to run:
```python
qiime2.utils.duplicate(file_in, file_out)
```
Which will hard-link if the files are on the same device (instant), and otherwise, copy byte-wise with a reasonable buffer size (as fast as anything else), avoiding the line-processing (which is what's actually pretty slow, especially readlines())

Reply:
Actually, given the next step, we can let the gzip module handle it (since we have to rewrite the data no matter what):

https://docs.python.org/3/library/gzip.html#examples-of-usage

If you were to open `file_out` via `gzip`, then it will compress the stream as it writes in 8k blocks (through         `shutil.copyfileobj(f_in, f_out)`).
You would also open `file_in` in `rb` mode so that no text-processing is done (there's no reason to use the unicode decoder in the middle of the file-handle stack).